### PR TITLE
Allow blockquotes w/ nested paragraphs to keep markup

### PR DESCRIPTION
**Pull request by aileen**
no issue

- blockquotes with nested paragraph children are valid HTML
- the current behaviour in Mobiledoc is to create a separate `<blockquote>` sections for each nested paragraph
- this behaviour is not desired when running migrations from other HTML sources, as it breaks the intended semantics
- this change will insert `soft-returns` between paragraph children and keep the `innerHtml` of those, so the default upstream logic doesn't grab
- only blockquotes with multiple children and sequential paragraphs (`p + p`) will be processed by this change
- this fix should probably be added into the upstream dependency and update the current behaviour which was created [here](https://github.com/bustle/mobiledoc-kit/issues/715)

### CODE REVIEW

**Review by kevinansfield — CHANGES_REQUESTED** [review]
Could you add a test that has non-paragraph block-level elements inside a blockquote? `<h1>` within a blockquote should create a new header section rather than text. I've not tested but only using `child.innerHTML` would strip the semantic elements

**Comment by kevinansfield** [comment]
A test for markup elements inside a blockquote would be useful too 🙂 eg, `<blockquote><a href="https://test.com">Content</a></blockquote>`

**Comment by aileen** [comment]
Yeah, makes sense. I'm changing it now to really only tackle sequential paragraphs. I think it might mess up with all other results otherwise, so only checking for `p + p` situations. Having some trouble testing it properly, as I'm still kinda new to the structure and trying to figure out what I need to expect. Using the tests [here](https://github.com/bustle/mobiledoc-kit/blob/b7e109b561e8400164aa7478058166c14bb0703a/src/js/parsers/section.ts) for orientation.

**Comment by aileen** [comment]
Ok, figured it out with the test. All I needed was a night of sleep, lol

**Comment by aileen** [comment]
@kevinansfield ping 😃

File: packages/kg-parser-plugins/lib/parser-plugins.js [hunk 1]
```diff
@@ -510,6 +510,38 @@ export function createParserPlugins(_options = {}) {
...
+
+        children.forEach((child) => {
+            let nextSibling = child.nextSibling;
+            let previousSibling = child.previousSibling;
+
+            // Only add a soft-break for two sequential paragraphs.
+            // Use the innerHTML only in that case, so Mobiledoc's default behaviour
+            // of creating separate blockquotes doesn't apply.
+            if (child.tagName === 'P' && (nextSibling && nextSibling.tagName === 'P')) {
+                html.push(`${child.innerHTML}<br>`);
```
Comment (kevinansfield) [inline packages/kg-parser-plugins/lib/parser-plugins.js:532]: Would `<br><br>` be more appropriate here to better match the intended display of separate paragraphs?

Comment (aileen) [reply packages/kg-parser-plugins/lib/parser-plugins.js]: Yup, makes sense! Just updated. Pasting html like this:

```html
<blockquote><p>My first quote line</p><p>My second quote line</p><p>My third quote line</p><p>My fourth quote line</p></blockquote>
```

results in this now:
image

**Comment by aileen** [comment]
@kevinansfield can you check if this is ok now?

**Review by kevinansfield — APPROVED** [review]
👍 LGTM

**Comment by aileen** [comment]
cc @PaulAdamDavis we need to update the migration tool as soon as this change is published in a new version.

**Comment by kevinansfield** [comment]
@aileen @PaulAdamDavis this was just shipped as `1.1.0`

**Comment by PaulAdamDavis** [comment]
Updated in https://github.com/TryGhost/migrate/commit/01e87bfe332579b24fbc3854e756994c2e599964

**Pull request merged**